app/controllers/policy_assessments_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PolicyAssessmentsController < PlanningApplicationsController
+  before_action :set_planning_application
+  before_action :set_policy_class, only: %i[show_class update_class]
+
+  def part; end
+
+  def new
+    @part = params[:part]
+
+    redirect_to part_new_policy_assessment_path(@planning_application), alert: "Please choose one of the policy parts" if @part.blank?
+  end
+
+  def create
+    class_ids = policy_class_params[:policy_classes].reject(&:blank?)
+
+    classes = PolicyClass
+                .classes_for_part(params[:part])
+                .select { |c| class_ids.include? c.id }
+                .each { |c| c.stamp_status! }
+
+    @planning_application.policy_classes += classes
+
+    if @planning_application.save
+      redirect_to @planning_application, notice: "classes successfully added"
+    else
+      render :new
+    end
```

Comment:
This action and the update_class one feel a little clunky but they can be iterated on.

Reply:
Agreed; I ended up writing the specs later, but once they're all green I'll start refactoring